src/_macosx.m

```diff
@@ -281,6 +281,9 @@ static void lazy_init(void) {
     NSApp = [NSApplication sharedApplication];
     [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
 
+    NSImage* icon = [[NSImage alloc]initWithContentsOfFile:@"/Users/dstansby/github/matplotlib/lib/matplotlib/mpl-data/images/matplotlib_large.png"];
```

Comment:
The next thing is to work out how to get this absolute path properly instead of hardcoding it for my laptop! Does anyone have any ideas?

Reply:
Perhaps something like (untested)
```c
PyObject* mpl = NULL, data_path_str = NULL, data_path_bytes = NULL;
char* data_path;
if (!(mpl = PyImport_ImportModule("matplotlib")) ||
    !(data_path_str = PyObject_CallMethod(mpl, "get_data_path", NULL)) ||
    !(data_path_bytes = PyUnicode_EncodeFSDefault(data_path_str)) ||
    !(data_path = PyBytes_AsString(data_path_bytes))) {
    goto exit;
}
// at that point data_path contains everything up to mpl-data,
// then you need to concatenate whatever e.g. with strcat.
exit:
Py_XDECREF(mpl);
Py_XDECREF(data_path_str);
Py_XDECREF(data_path_bytes);
```

Reply:
Can we not pass it in somehow, like done with navigation toolbar icons?